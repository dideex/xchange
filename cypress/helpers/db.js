const mongoose = require('mongoose')
const mongoConfig = require('./config.json').mongodb

module.exports = mongoose.connect(mongoConfig, {useNewUrlParser: true})
