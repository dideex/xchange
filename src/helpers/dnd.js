import React from 'react'
import {DragDropContext} from 'react-dnd'
import TestBackend from 'react-dnd-test-backend'

export function wrapInTestContext(DecoratedComponent) {
  const TestStub = props => <DecoratedComponent {...props} />
  return DragDropContext(TestBackend)(TestStub)
}
