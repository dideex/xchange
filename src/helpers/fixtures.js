export const fakeCurrnecy = [
  {
    order: 1,
    _id: 1,
    icon: 'eth',
    id: 'eth',
    name: 'tether',
    label: 'tether label',
    source: 'tether source wallet'
  },
  {
    order: 2,
    _id: 2,
    icon: 'bitcoin',
    id: 'bitcoin',
    name: 'lisk',
    label: 'lisk label',
    source: 'lisk source wallet'
  },
  {
    order: 3,
    _id: 3,
    icon: 'rur',
    id: 'rur',
    name: 'litecoin',
    label: 'litecoin label',
    source: 'litecoin source wallet'
  },
]


export const fakeData = {
  id: 'Test id',
  inputValue: '1000',
  toWallet: 'Test wallet',
  currencyInputLabel: 'Test input label',
  currencyOutputLabel: 'Test output label',
  fromWallet: 'Tests fromWallet',
  outputValue: '2000',
  email: 'Tests email',
  username: 'Tests username',
  paymentStatus: 1,
  loading: false,
  wallets: {
    eth: 'eth wallet',
    bitcoint: 'bitcoint wallet',
    rur: 'rur wallet',
  },
  login: 'Test login',
  updatePaymentStatus() {},
}

export const fakeUser = {
  username: 'Fake username',
  password: 'Fake password',
  email: 'fake@email.com',
  phone: '89145678901',
  fio: 'Fake fio',
  token: 'Fake token',
}

export const fakeBitcoinAddress = '3JQSigWTCHyBLRD979JWgEtWP5YiiFwcQB '
export const fakeCreditAddress = '1234 4321 5678 8765'

export const fakeWallets = {
  eth: 'Fake eth wallet',
  bitcoin: 'Fake bitcoin wallet',
  rur: 'Fake rur wallet'
}