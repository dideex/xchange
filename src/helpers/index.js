export {fakeCurrnecy} from './fixtures'

export const delay = (fn = () => {}, timeout = 0) =>
  new Promise(res =>
    setTimeout(() => {
      fn()
      res()
    }, timeout),
  )
