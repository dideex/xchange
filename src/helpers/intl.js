import React from 'react'
import {mount, shallow} from 'enzyme'
import {IntlProvider, intlShape} from 'react-intl'

import messages from '../locale/en.json'
const intlProvider = new IntlProvider({locale: 'en', messages}, {})
const {intl} = intlProvider.getChildContext()

function nodeWithIntlProp(node) {
  return React.cloneElement(node, {intl})
}

export function mountWithIntl(
  node,
  {context, childContextTypes, ...additionalOptions} = {},
) {
  return mount(nodeWithIntlProp(node), {
    context: Object.assign({}, context, {intl}),
    childContextTypes: Object.assign({}, {intl: intlShape}, childContextTypes),
    ...additionalOptions,
  })
}
