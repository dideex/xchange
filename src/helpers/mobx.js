import React from 'react'
import {Provider} from 'mobx-react'
import {lastOperations, cash, user} from '../store'

jest.mock('../components/Api', () => ({
  post: () => Promise.resolve({data: {}}),
  get: () => Promise.resolve({data: {}}),
  errorEmitter: jest.fn(data => fn => data(fn)),
}))

// jest.mock('../components/common', () => ({
//   noty: () => {},
//   Loading: () => <div>Loading</div>,
// }))

export const MobxProvider = ({children, ...stores}) => {
  return (
    <Provider
      lastOperationsStore={lastOperations}
      cashStore={cash}
      userStore={user}
      {...stores}
    >
      {children}
    </Provider>
  )
}
