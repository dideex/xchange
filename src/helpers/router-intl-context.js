import {mount} from 'enzyme'
import {BrowserRouter} from 'react-router-dom'
import {IntlProvider, intlShape} from 'react-intl'
import {shape} from 'prop-types'

import messages from '../locale/en.json'

export const testId = 'test id'

// Instantiate router context
const router = {
  history: {...new BrowserRouter().history},
  route: {
    location: {
      path: '/location',
      href: '/location',
    },
    match: {
      path: '/location',
      params: {
        id: testId,
      },
    },
  },
}

// const createContext = () => ({
//   context: {router},
//   childContextTypes: {router: shape({})},
// })

const intlProvider = new IntlProvider({locale: 'en', messages}, {})
const {intl} = intlProvider.getChildContext()

const createMixedContext = ({
  router,
  context,
  childContextTypes,
  ...additionalOptions
} = {}) => ({
  context: Object.assign({}, {router}, context, {intl}),
  childContextTypes: Object.assign(
    {},
    {router: shape({})},
    {intl: intlShape},
    childContextTypes,
  ),
  ...additionalOptions,
})

export function mountWrap(node) {
  return mount(node, createMixedContext({router}))
}

// export function shallowWrap(node) {
//   return shallow(node, createContext())
// }
