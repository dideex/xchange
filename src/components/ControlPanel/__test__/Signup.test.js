import React from 'react'
import Component from '../Signup'
import {Provider as MobxProvider} from 'mobx-react'

import UserStore from '../../../store/User'

import {mountWrap} from '../../../helpers/router-intl-context'
import {fakeData, fakeUser} from '../../../helpers/fixtures'
import {delay} from '../../../helpers'

jest.mock('../../../components/Api', () => ({
  post: () => Promise.resolve({data: {}}),
  get: () => Promise.resolve({data: {}, eth: {}}),
  errorEmitter: fn => data => fn(data),
}))

jest.mock('../../../components/common/Noty.js', () => ({
  noty: () => {},
}))

describe('Control panel: signin', () => {
  let userStore
  const fakeLogin = 'Fake login'
  beforeEach(() => {
    userStore = new UserStore()
    userStore.fetchGuestOrder = () => Promise.resolve({...fakeData, paymentStatus: 1})
  })

  describe('Component markup', () => {
    it('Basic markup', () => {
      const wrapper = mountWrap(
        <MobxProvider userStore={userStore}>
          <Component />
        </MobxProvider>,
      )
      expect(wrapper.html()).toMatchSnapshot()
    })
  })

  describe('Component methods bahavriour', () => {
    it('Should redirect when token is exist', () => {
      userStore.token = true
      const wrapper = mountWrap(
        <MobxProvider userStore={userStore}>
          <Component />
        </MobxProvider>,
      )
      // FIXME: add history.push observer
      expect(wrapper.html()).toMatchSnapshot()
    })

    it('Errors should be cleared after cDM', () => {
      userStore.clearErr = jest.fn()
      const wrapper = mountWrap(
        <MobxProvider userStore={userStore}>
          <Component />
        </MobxProvider>,
      )
      expect(userStore.clearErr).toHaveBeenCalledTimes(1)
    })

    it('Window should scroll to the header after cDM', () => {
      const {scrollTo} = window
      window.scrollTo = jest.fn()
      const wrapper = mountWrap(
        <MobxProvider userStore={userStore}>
          <Component />
        </MobxProvider>,
      )
      expect(window.scrollTo).toHaveBeenCalledTimes(1)
      window.scrollTo = scrollTo
    })

    it('Password should be cleaned after cWU', () => {
      userStore.changePassword = jest.fn()
      const wrapper = mountWrap(
        <MobxProvider userStore={userStore}>
          <Component />
        </MobxProvider>,
      )
      wrapper.unmount()
      expect(userStore.changePassword).toHaveBeenCalledTimes(1)
      expect(userStore.changePassword).toHaveBeenCalledWith('')
    })

    it('Handle submit should not work with untouched inputs', async () => {
      userStore.signupUser = jest.fn()
      const wrapper = mountWrap(
        <MobxProvider userStore={userStore}>
          <Component />
        </MobxProvider>,
      )

      const instance = wrapper.find('SignUp').instance()
      instance.handleSubmit()
      await delay()
      expect(userStore.signupUser).toHaveBeenCalledTimes(0)
    })

    it('Handle submit should not work with empty inputs', async () => {
      userStore.signupUser = jest.fn()
      const wrapper = mountWrap(
        <MobxProvider userStore={userStore}>
          <Component />
        </MobxProvider>,
      )

      wrapper
        .find('SignUp')
        .setState({usernameError: true, passwordError: true, emailError: true})
      const instance = wrapper.find('SignUp').instance()
      instance.handleSubmit()
      await delay()
      expect(userStore.signupUser).toHaveBeenCalledTimes(0)
    })

    it('Handle submit should work', async () => {
      userStore.signupUser = jest.fn()
      const wrapper = mountWrap(
        <MobxProvider userStore={userStore}>
          <Component />
        </MobxProvider>,
      )

      userStore.changeLogin(fakeUser.username)
      userStore.changeUsername(fakeUser.username)
      userStore.changePassword(fakeUser.password)
      userStore.changeEmail(fakeUser.email)
      wrapper.find('SignUp').setState({
        usernameError: false,
        passwordError: false,
        emailError: false,
        passwordRepeated: fakeUser.password,
      })
      wrapper
        .find('SignUp')
        .instance()
        .handleSubmit()
      await delay()
      expect(userStore.signupUser).toHaveBeenCalledTimes(1)
    })
  })

  describe('Form bahavriour', () => {
    it('Login field should change', () => {
      const wrapper = mountWrap(
        <MobxProvider userStore={userStore}>
          <Component />
        </MobxProvider>,
      )
      wrapper
        .find('input[placeholder="Login"]')
        .simulate('change', {target: {value: fakeLogin}})
      expect(userStore.login).toBe(fakeLogin)
      expect(wrapper.find('input[placeholder="Login"]').instance().value).toBe(fakeLogin)
    })

    it('Username field should change', () => {
      const wrapper = mountWrap(
        <MobxProvider userStore={userStore}>
          <Component />
        </MobxProvider>,
      )
      wrapper
        .find('input[placeholder="Full name"]')
        .simulate('change', {target: {value: fakeUser.fio}})
      expect(userStore.username).toBe(fakeUser.fio)
      expect(wrapper.find('input[placeholder="Full name"]').instance().value).toBe(
        fakeUser.fio,
      )
    })

    it('Email field should change', () => {
      const wrapper = mountWrap(
        <MobxProvider userStore={userStore}>
          <Component />
        </MobxProvider>,
      )
      wrapper
        .find('input[placeholder="Email"]')
        .simulate('change', {target: {value: fakeUser.email}})
      expect(userStore.email).toBe(fakeUser.email)
      expect(wrapper.find('input[placeholder="Email"]').instance().value).toBe(
        fakeUser.email,
      )
    })

    it('Password field should change', () => {
      const wrapper = mountWrap(
        <MobxProvider userStore={userStore}>
          <Component />
        </MobxProvider>,
      )
      wrapper
        .find('input[placeholder="Password"]')
        .simulate('change', {target: {value: fakeUser.password}})
      expect(userStore.password).toBe(fakeUser.password)
      expect(wrapper.find('input[placeholder="Password"]').instance().value).toBe(
        fakeUser.password,
      )
    })

    it('Repeat password field should change', () => {
      const wrapper = mountWrap(
        <MobxProvider userStore={userStore}>
          <Component />
        </MobxProvider>,
      )
      wrapper
        .find('input[placeholder="Repeat password"]')
        .simulate('change', {target: {value: fakeUser.password}})
      expect(
        wrapper
          .find('SignUp')
          .instance()
          .state.passwordRepeated,
      ).toBe(fakeUser.password)
      expect(wrapper.find('input[placeholder="Repeat password"]').instance().value).toBe(
        fakeUser.password,
      )
    })

    
    it('Error fields should appear', () => {
      const wrapper = mountWrap(
        <MobxProvider userStore={userStore}>
          <Component />
        </MobxProvider>,
      )

      wrapper.find('button').simulate('click')
      expect(wrapper.html()).toMatchSnapshot()
    })

    it('Error fields should hide when problem solves', async () => {
      const wrapper = mountWrap(
        <MobxProvider userStore={userStore}>
          <Component />
        </MobxProvider>,
      )

      wrapper.find('button').simulate('click')
      wrapper
        .find('input[placeholder="Full name"]')
        .simulate('change', {target: {value: fakeUser.fio}})
      wrapper
        .find('input[placeholder="Email"]')
        .simulate('change', {target: {value: fakeUser.email}})
      wrapper
        .find('input[placeholder="Password"]')
        .simulate('change', {target: {value: fakeUser.password}})
      wrapper
        .find('input[placeholder="Repeat password"]')
        .simulate('change', {target: {value: fakeUser.password}})

      await delay(() => {}, 500)
      expect(wrapper.html()).toMatchSnapshot()
    })
  })
})
