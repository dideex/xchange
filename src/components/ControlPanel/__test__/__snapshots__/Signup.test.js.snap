// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Control panel: signin Component markup Basic markup 1`] = `"<div class=\\"css-18nqoil\\"><div><h2 class=\\"css-x184e\\"><span>Sign up</span></h2></div><label class=\\"css-gudk1o\\"><input placeholder=\\"Login\\" pattern=\\"[a-zа-яё]{2,}\\" type=\\"text\\" mask=\\"\\" class=\\"css-18b8ap6\\" value=\\"\\"></label><label class=\\"css-gudk1o\\"><input placeholder=\\"Full name\\" type=\\"text\\" mask=\\"\\" class=\\"css-18b8ap6\\" value=\\"\\"></label><label class=\\"css-gudk1o\\"><input placeholder=\\"Email\\" pattern=\\"^(([^<>()\\\\[\\\\]\\\\.,;:\\\\s@&quot;]+(\\\\.[^<>()\\\\[\\\\]\\\\.,;:\\\\s@&quot;]+)*)|(&quot;.+&quot;))@((\\\\[[0-9]{1,3}\\\\.[0-9]{1,3}\\\\.[0-9]{1,3}\\\\.[0-9]{1,3}\\\\])|(([a-zA-Z\\\\-0-9]+\\\\.)+[a-zA-Z]{2,}))$\\" type=\\"text\\" mask=\\"\\" class=\\"css-18b8ap6\\" value=\\"\\"></label><label class=\\"css-gudk1o\\"><input type=\\"password\\" pattern=\\"\\\\S{6,}\\" placeholder=\\"Password\\" mask=\\"\\" class=\\"css-18b8ap6\\" value=\\"\\"></label><label class=\\"css-gudk1o\\"><input type=\\"password\\" pattern=\\"\\\\S{6,}\\" placeholder=\\"Repeat password\\" mask=\\"\\" class=\\"css-18b8ap6\\" value=\\"\\"></label><button cursor=\\"pointer\\" class=\\"css-8hkfp7\\"><div class=\\"css-1v1d0dg\\">Sign up</div></button></div>"`;

exports[`Control panel: signin Component methods bahavriour Should redirect when token is exist 1`] = `"<div class=\\"css-18nqoil\\"><div><h2 class=\\"css-x184e\\"><span>Sign up</span></h2></div><label class=\\"css-gudk1o\\"><input placeholder=\\"Login\\" pattern=\\"[a-zа-яё]{2,}\\" type=\\"text\\" mask=\\"\\" class=\\"css-18b8ap6\\" value=\\"\\"></label><label class=\\"css-gudk1o\\"><input placeholder=\\"Full name\\" type=\\"text\\" mask=\\"\\" class=\\"css-18b8ap6\\" value=\\"\\"></label><label class=\\"css-gudk1o\\"><input placeholder=\\"Email\\" pattern=\\"^(([^<>()\\\\[\\\\]\\\\.,;:\\\\s@&quot;]+(\\\\.[^<>()\\\\[\\\\]\\\\.,;:\\\\s@&quot;]+)*)|(&quot;.+&quot;))@((\\\\[[0-9]{1,3}\\\\.[0-9]{1,3}\\\\.[0-9]{1,3}\\\\.[0-9]{1,3}\\\\])|(([a-zA-Z\\\\-0-9]+\\\\.)+[a-zA-Z]{2,}))$\\" type=\\"text\\" mask=\\"\\" class=\\"css-18b8ap6\\" value=\\"\\"></label><label class=\\"css-gudk1o\\"><input type=\\"password\\" pattern=\\"\\\\S{6,}\\" placeholder=\\"Password\\" mask=\\"\\" class=\\"css-18b8ap6\\" value=\\"\\"></label><label class=\\"css-gudk1o\\"><input type=\\"password\\" pattern=\\"\\\\S{6,}\\" placeholder=\\"Repeat password\\" mask=\\"\\" class=\\"css-18b8ap6\\" value=\\"\\"></label><button cursor=\\"pointer\\" class=\\"css-8hkfp7\\"><div class=\\"css-1v1d0dg\\">Sign up</div></button></div>"`;

exports[`Control panel: signin Form bahavriour Error fields should appear 1`] = `"<div class=\\"css-18nqoil\\"><div><h2 class=\\"css-x184e\\"><span>Sign up</span></h2></div><label class=\\"css-gudk1o\\"><input placeholder=\\"Login\\" pattern=\\"[a-zа-яё]{2,}\\" type=\\"text\\" mask=\\"\\" class=\\"css-ykrnj4\\" value=\\"\\"><span class=\\"css-1j8ju8e error--enter error--enter-active\\">Enter your full name</span></label><label class=\\"css-gudk1o\\"><input placeholder=\\"Full name\\" type=\\"text\\" mask=\\"\\" class=\\"css-ykrnj4\\" value=\\"\\"><span class=\\"css-1j8ju8e error--enter error--enter-active\\">Enter your login</span></label><label class=\\"css-gudk1o\\"><input placeholder=\\"Email\\" pattern=\\"^(([^<>()\\\\[\\\\]\\\\.,;:\\\\s@&quot;]+(\\\\.[^<>()\\\\[\\\\]\\\\.,;:\\\\s@&quot;]+)*)|(&quot;.+&quot;))@((\\\\[[0-9]{1,3}\\\\.[0-9]{1,3}\\\\.[0-9]{1,3}\\\\.[0-9]{1,3}\\\\])|(([a-zA-Z\\\\-0-9]+\\\\.)+[a-zA-Z]{2,}))$\\" type=\\"text\\" mask=\\"\\" class=\\"css-ykrnj4\\" value=\\"\\"><span class=\\"css-1j8ju8e error--enter error--enter-active\\">Enter your email</span></label><label class=\\"css-gudk1o\\"><input type=\\"password\\" pattern=\\"\\\\S{6,}\\" placeholder=\\"Password\\" mask=\\"\\" class=\\"css-ykrnj4\\" value=\\"\\"><span class=\\"css-1j8ju8e error--enter error--enter-active\\">Enter your password</span></label><label class=\\"css-gudk1o\\"><input type=\\"password\\" pattern=\\"\\\\S{6,}\\" placeholder=\\"Repeat password\\" mask=\\"\\" class=\\"css-18b8ap6\\" value=\\"\\"></label><button cursor=\\"pointer\\" class=\\"css-8hkfp7\\"><div class=\\"css-1v1d0dg\\">Sign up</div></button></div>"`;

exports[`Control panel: signin Form bahavriour Error fields should hide when problem solves 1`] = `"<div class=\\"css-18nqoil\\"><div><h2 class=\\"css-x184e\\"><span>Sign up</span></h2></div><label class=\\"css-gudk1o\\"><input placeholder=\\"Login\\" pattern=\\"[a-zа-яё]{2,}\\" type=\\"text\\" mask=\\"\\" class=\\"css-ykrnj4\\" value=\\"\\"><span class=\\"css-1j8ju8e error--enter-done\\">Enter your full name</span></label><label class=\\"css-gudk1o\\"><input placeholder=\\"Full name\\" type=\\"text\\" mask=\\"\\" class=\\"css-18b8ap6\\" value=\\"Fake fio\\"></label><label class=\\"css-gudk1o\\"><input placeholder=\\"Email\\" pattern=\\"^(([^<>()\\\\[\\\\]\\\\.,;:\\\\s@&quot;]+(\\\\.[^<>()\\\\[\\\\]\\\\.,;:\\\\s@&quot;]+)*)|(&quot;.+&quot;))@((\\\\[[0-9]{1,3}\\\\.[0-9]{1,3}\\\\.[0-9]{1,3}\\\\.[0-9]{1,3}\\\\])|(([a-zA-Z\\\\-0-9]+\\\\.)+[a-zA-Z]{2,}))$\\" type=\\"text\\" mask=\\"\\" class=\\"css-18b8ap6\\" value=\\"fake@email.com\\"></label><label class=\\"css-gudk1o\\"><input type=\\"password\\" pattern=\\"\\\\S{6,}\\" placeholder=\\"Password\\" mask=\\"\\" class=\\"css-18b8ap6\\" value=\\"Fake password\\"></label><label class=\\"css-gudk1o\\"><input type=\\"password\\" pattern=\\"\\\\S{6,}\\" placeholder=\\"Repeat password\\" mask=\\"\\" class=\\"css-18b8ap6\\" value=\\"Fake password\\"></label><button cursor=\\"pointer\\" class=\\"css-8hkfp7\\"><div class=\\"css-1v1d0dg\\">Sign up</div></button></div>"`;
